http-scala-fx/src/main/scala/fx/Http.scala

```diff
@@ -0,0 +1,319 @@
+package fx
+
+import java.net.URI
+import java.net.http.HttpHeaders
+import java.net.http.HttpRequest
+import java.net.http.HttpRequest.BodyPublishers
+import java.net.http.HttpResponse
+import java.net.http.HttpResponse.BodyHandlers
+import java.time.Duration
+import java.util.Random
+import scala.annotation.tailrec
+import scala.concurrent.duration.{Duration => ScalaDuration}
+
+/**
+ * Models the http scheme
+ */
+type Http[A] = (
+    HttpRetryPolicy[A],
+    Control[HttpExecutionException],
+    HttpClient
+) ?=> A
+
+extension [A](http: Http[A])
+  def httpValue(using Control[HttpExecutionException]): A = http
+  def fmap[B](f: A => B): Http[B] =
+    f(http)
+  def bindMap[B](f: A => Http[B]): Http[B] =
+    f(http)
+
+extension (uri: URI)
+  private def addHeadersToRequestBuilder(
+      builder: HttpRequest.Builder,
+      headers: HttpHeader*): HttpRequest.Builder =
+    headers.foldLeft(builder) { (builder: HttpRequest.Builder, header: HttpHeader) =>
+      header.values.foldLeft[HttpRequest.Builder](builder) {
+        (builder: HttpRequest.Builder, headerValue: String) =>
+          builder.header(header.name, headerValue)
+      }
+    }
+
+  def GET[A]()(using HttpResponseMapper[A]): Http[HttpResponse[A]] =
+    GET(HttpRetries(0), None, List.empty: _*)
+
+  def GET[A](headers: HttpHeader*)(using HttpResponseMapper[A]): Http[HttpResponse[A]] =
+    GET(HttpRetries(0), None, headers: _*)
+
+  def GET[A](timeout: ScalaDuration, headers: HttpHeader*)(
+      using HttpResponseMapper[A]): Http[HttpResponse[A]] =
+    GET(HttpRetries(0), Option(timeout), headers: _*)
+
+  @tailrec
+  private def GET[A](
+      retryCount: HttpRetries,
+      maybeTimeout: Option[ScalaDuration],
+      headers: HttpHeader*)(using HttpResponseMapper[A]): Http[HttpResponse[A]] =
+    val mapper = summon[HttpResponseMapper[A]]
```

Comment:
An alternative here to `summon` is named args (using mapper: HttpResponseMapper[A]). No strong opinion though.

Reply:
Yeah, I know. Thought I caught all those, but I‌ apparently didn't. Resolved in [a21be53](https://github.com/47deg/TBD/pull/37/commits/a21be5315c0fa0696a8954dee9026ac71e03c5ed).